### PR TITLE
Add Recipe multiplier

### PR DESCRIPTION
**Pull request by mxdvl**
## What does this change?

Hack day project to help multiply quantities in recipes.

## Why?

Sometimes you want to cook more or less.

## Screenshots

image

### CODE REVIEW

File: dotcom-rendering/babel.config.js [hunk 1]
```diff
@@ -8,6 +8,7 @@ module.exports = {
 		'babel-plugin-px-to-rem',
 		'@babel/plugin-transform-runtime',
 		'@emotion/babel-plugin',
+		["polyfill-corejs3", { "method": "usage-global", "version": "3.20" }]
```
Comment (mxdvl) [inline dotcom-rendering/babel.config.js:11]: This is needed for `String.matchAll`.

File: dotcom-rendering/index.d.ts [hunk 2]
```diff
@@ -565,7 +565,7 @@ declare namespace JSX {
 	interface IntrinsicElements {
 		'gu-island': {
 			name: string;
-			deferUntil?: 'idle' | 'visible' | 'interaction';
+			deferUntil?: 'idle' | 'visible' | 'interaction' | 'feature';
```
Comment (mxdvl) [inline dotcom-rendering/index.d.ts:568]: This may not be apltly named…

**Review by mxdvl — COMMENTED** [review]
Some pointers

File: dotcom-rendering/index.d.ts (hunk 2 above)
Comment (bryophyta) [reply dotcom-rendering/index.d.ts:568]: Perhaps: `defer-until="feature-flag"`?

Comment (bryophyta) [reply dotcom-rendering/index.d.ts]: just seen that you went for `hash` -- that sounds right to me; good to be explicit!

**Review by tkgnm — APPROVED** [review]
🚢 !!!

File: dotcom-rendering/src/web/lib/renderElement.tsx [hunk 3]
```diff
@@ -777,7 +777,7 @@ export const renderElement = ({
 // bareElements is the set of element types that don't get wrapped in a Figure
 // for most article types, either because they don't need it or because they
 // add the figure themselves.
-const bareElements = new Set([
+const bareElements = new Set<CAPIElement['_type']>([
```
Comment (mxdvl) [inline dotcom-rendering/src/web/lib/renderElement.tsx:780]: This is just more explicit about what this set contains!

**Review by mxdvl — COMMENTED** [review]
Some pointers for future travellers, as we plan to put this into PROD

File: dotcom-rendering/cypress/support/commands.js [hunk 4]
```diff
@@ -28,7 +28,7 @@ import 'cypress-wait-until';
 
 Cypress.Commands.add('hydrate', () => {
 	return cy
-		.get('gu-island')
+		.get('gu-island[deferUntil=visible]')
```
[the remark this answers is not in the document]
Comment (arelra) [reply dotcom-rendering/cypress/support/commands.js:31]: Nice hack! 👏 

This was a catch-all to make sure islands of all types are 'ready' to take a visual snapshot.

You're right that non 'visible' islands don't need to scroll but we'd need an equivalent wait for each island type.

I think this command is only used by Percy so it shouldn't fail the main CI.

[the remark this answers is not in the document]
Comment (mxdvl) [reply dotcom-rendering/cypress/support/commands.js:31]: Well… it was blocking us, so we needed to find a way 😆

**Pull request merged**